src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsHostBuilderExtensions.cs

```diff
@@ -57,52 +60,57 @@ public static IHostBuilder AddApplicationInsights(this IHostBuilder builder, str
                 {
                     ITelemetryChannel channel = provider.GetService<ITelemetryChannel>();
-                    TelemetryConfiguration config = new TelemetryConfiguration(instrumentationKey, channel);
-                    
-                    foreach (ITelemetryInitializer initializer in provider.GetServices<ITelemetryInitializer>())
-                    {
-                        config.TelemetryInitializers.Add(initializer);
-                    }
-
-                    QuickPulseTelemetryModule quickPulseModule = null;
-                    foreach (ITelemetryModule module in provider.GetServices<ITelemetryModule>())
-                    {
-                        if (module is QuickPulseTelemetryModule telemetryModule)
-                        {
-                            quickPulseModule = telemetryModule;
-                        }
-                        module.Initialize(config);
-                    }
-
-                    QuickPulseTelemetryProcessor processor = null;
-                    config.TelemetryProcessorChainBuilder
-                        .Use((next) =>
-                        {
-                            processor = new QuickPulseTelemetryProcessor(next);
-                            return processor;
-                        })
-                        .Use((next) => new FilteringTelemetryProcessor(filter, next));
+                    TelemetryConfiguration config = TelemetryConfiguration.CreateDefault();
+                    SetupTelemetryConfiguration(
+                        config,
+                        instrumentationKey, 
+                        filter, 
+                        samplingSettings,
+                        channel,
+                        provider.GetServices<ITelemetryInitializer>(),
+                        provider.GetServices<ITelemetryModule>());
 
-                    if (samplingSettings != null)
+                    // Function users have no access to TelemetryConfiguration from host DI container,
+                    // so we'll expect user to work with TelemteryConfiguration.Active
+                    // Also, some ApplicaitonInsights internal operations (heartbeats) depend on
+                    // the TelemetryConfiguration.Active being set so, we'll set up Active once per process lifetime.
+                    TelemetryConfiguration activeConfig = TelemetryConfiguration.Active;
+                    if (string.IsNullOrEmpty(activeConfig.InstrumentationKey))
                     {
-                        config.TelemetryProcessorChainBuilder.Use((next) =>
-                            new AdaptiveSamplingTelemetryProcessor(samplingSettings, null, next));
+                        SetupTelemetryConfiguration(
+                            activeConfig,
+                            instrumentationKey,
+                            filter,
+                            samplingSettings,
+                            new ServerTelemetryChannel(),
```

Comment:
Is this meant to get a new instance of the channel? Or can it use the same as the other config?

Reply:
This meant to have a new ServerTelemetryChannel. As host could be disposed, it will cause TelemtryConfiguration disposal and the underlying channel will also be disposed. 
If channel is reused TelemetryConfiguration.Active becomes broken after host dies, so it has to have another instance which lifetime is process lifetime.